xbmc/cores/DataCacheCore.cpp

```diff
@@ -354,6 +358,30 @@ void CDataCacheCore::GetPlayTimes(time_t &start, int64_t &current, int64_t &min,
   max = m_timeInfo.m_timeMax;
 }
 
+void CDataCacheCore::SetCutList(const std::vector<EDL::Cut>& cutList)
+{
+  CSingleLock lock(m_stateSection);
```

Comment:
stateSection protects player states. Cut list and chapters are not player states.

Reply:
Which of the existing section members to use? or introduce a new one?

Reply:
would that be okay?

```
diff --git a/xbmc/cores/DataCacheCore.cpp b/xbmc/cores/DataCacheCore.cpp
index 46f8cd8223..9bd70afc25 100644
--- a/xbmc/cores/DataCacheCore.cpp
+++ b/xbmc/cores/DataCacheCore.cpp
@@ -177,6 +177,30 @@ float CDataCacheCore::GetVideoDAR()
   return m_playerVideoInfo.dar;
 }
 
+void CDataCacheCore::SetCutList(const std::vector<EDL::Cut>& cutList)
+{
+  CSingleLock lock(m_videoPlayerSection);
+  m_playerVideoInfo.m_cutList = cutList;
+}
+
+std::vector<EDL::Cut> CDataCacheCore::GetCutList() const
+{
+  CSingleLock lock(m_videoPlayerSection);
+  return m_playerVideoInfo.m_cutList;
+}
+
+void CDataCacheCore::SetChapters(const std::vector<std::pair<std::string, int64_t>>& chapters)
+{
+  CSingleLock lock(m_videoPlayerSection);
+  m_playerVideoInfo.m_chapters = chapters;
+}
+
+std::vector<std::pair<std::string, int64_t>> CDataCacheCore::GetChapters() const
+{
+  CSingleLock lock(m_videoPlayerSection);
+  return m_playerVideoInfo.m_chapters;
+}
+
 // player audio info
 void CDataCacheCore::SetAudioDecoderName(std::string name)
 {
@@ -358,30 +382,6 @@ void CDataCacheCore::GetPlayTimes(time_t &start, int64_t &current, int64_t &min,
   max = m_timeInfo.m_timeMax;
 }
 
-void CDataCacheCore::SetCutList(const std::vector<EDL::Cut>& cutList)
-{
-  CSingleLock lock(m_stateSection);
-  m_cutList = cutList;
-}
-
-std::vector<EDL::Cut> CDataCacheCore::GetCutList() const
-{
-  CSingleLock lock(m_stateSection);
-  return m_cutList;
-}
-
-void CDataCacheCore::SetChapters(const std::vector<std::pair<std::string, int64_t>>& chapters)
-{
-  CSingleLock lock(m_stateSection);
-  m_chapters = chapters;
-}
-
-std::vector<std::pair<std::string, int64_t>> CDataCacheCore::GetChapters() const
-{
-  CSingleLock lock(m_stateSection);
-  return m_chapters;
-}
-
 time_t CDataCacheCore::GetStartTime()
 {
   CSingleLock lock(m_stateSection);
diff --git a/xbmc/cores/DataCacheCore.h b/xbmc/cores/DataCacheCore.h
index 2861ed8192..11c96f1703 100644
--- a/xbmc/cores/DataCacheCore.h
+++ b/xbmc/cores/DataCacheCore.h
@@ -48,6 +48,10 @@ public:
   float GetVideoFps();
   void SetVideoDAR(float dar);
   float GetVideoDAR();
+  void SetCutList(const std::vector<EDL::Cut>& cutList);
+  std::vector<EDL::Cut> GetCutList() const;
+  void SetChapters(const std::vector<std::pair<std::string, int64_t>>& chapters);
+  std::vector<std::pair<std::string, int64_t>> GetChapters() const;
 
   // player audio info
   void SetAudioDecoderName(std::string name);
@@ -78,10 +82,6 @@ public:
   bool GetVideoRender();
   void SetPlayTimes(time_t start, int64_t current, int64_t min, int64_t max);
   void GetPlayTimes(time_t &start, int64_t &current, int64_t &min, int64_t &max);
-  void SetCutList(const std::vector<EDL::Cut>& cutList);
-  std::vector<EDL::Cut> GetCutList() const;
-  void SetChapters(const std::vector<std::pair<std::string, int64_t>>& chapters);
-  std::vector<std::pair<std::string, int64_t>> GetChapters() const;
 
   /*!
    * \brief Get the start time
@@ -127,7 +127,7 @@ public:
 protected:
   std::atomic_bool m_hasAVInfoChanges;
 
-  CCriticalSection m_videoPlayerSection;
+  mutable CCriticalSection m_videoPlayerSection;
   struct SPlayerVideoInfo
   {
     std::string decoderName;
@@ -139,6 +139,8 @@ protected:
     int height;
     float fps;
     float dar;
+    std::vector<EDL::Cut> m_cutList;
+    std::vector<std::pair<std::string, int64_t>> m_chapters; // name and position for chapters
   } m_playerVideoInfo;
 
   CCriticalSection m_audioPlayerSection;
@@ -156,7 +158,7 @@ protected:
     bool m_isClockSync;
   } m_renderInfo;
 
-  mutable CCriticalSection m_stateSection;
+  CCriticalSection m_stateSection;
   bool m_playerStateChanged = false;
   struct SStateInfo
   {
@@ -175,7 +177,4 @@ protected:
     int64_t m_timeMax;
     int64_t m_timeMin;
   } m_timeInfo = {};
-
-  std::vector<EDL::Cut> m_cutList;
-  std::vector<std::pair<std::string, int64_t>> m_chapters; // name and position for chapters
 };
```

Reply:
done

Reply:
I would create a new lock. playerVideoInfo is not about content, it's about process parameters used to play a given content. width/height for example are referring to video buffers not width/height of the content (though those can be the same). more clear when looking at i.e. deintMethod, which obviously does not belong to content.

Reply:
Okay, what would be a good name for the lock? m_contentSection?

Reply:
I added a new critical section `m_contentSection`.